src/classes/ADDR_AdminAcc_TEST.cls

```diff
@@ -0,0 +1,891 @@
+            system.assert(acc.BillingState.contains('Oregon')); 
+            system.assertEquals(false, acc.is_Address_Override__c);
+            system.assertNotEquals(null, acc.Current_Address__c);
+        }
+ 
+        // verify that no new address was created
+        list<Address__c> listAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Account__c from Address__c];
+        system.assertEquals(2, listAddr.size());
+        for (Address__c addr : listAddr) {
+            system.assertEquals('direct street  edit', addr.MailingStreet__c);
+        }        
+    }
+
+    /*********************************************************************************************************
+    @description Create an Address with no Account 
+    verify: runtime error
+    **********************************************************************************************************/            
+    @isTest
+    public static void newAddrNoParent() {
+        configSettings();
+        
+        List<Address__c> addrs = UTIL_UnitTestData_TEST.initTestAddr(2);
+        for (Integer i = 0; i < addrs.size(); i++) {
+            system.assertEquals(null, addrs[i].Parent_Account__c);
+        }
+        
+        try {
+            insert addrs;
+        } catch (Exception ex) {
+            system.assert(ex.getMessage().contains(Label.addrValidParentObjects));
+            return;
+        }
+        system.assert(false); //we shouldn't get here!
+    }
+ 
+    /*********************************************************************************************************
+    @description Create N new duplicate Addresses from Account  
+    verify:
+        duplicates not created
+    **********************************************************************************************************/            
+    @isTest
+    public static void newDupeAddr() {
+        configSettings();
+         
+        // this creates a default Address for each Account
+        UTIL_UnitTestData_TEST.AccsWithAddrs accsAddrs = UTIL_UnitTestData_TEST.createTestAccsAddrs(2, UTIL_Describe.getAdminAccRecTypeID());
+        List<Address__c> addrs = accsAddrs.addrs;
+        
+        // create additional addresses thru the account.
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            accsAddrs.accs[i].BillingStreet = addrs[i].MailingStreet__c;
+            accsAddrs.accs[i].BillingCity = addrs[i].MailingCity__c;
+            accsAddrs.accs[i].BillingState = addrs[i].MailingState__c;
+            accsAddrs.accs[i].BillingPostalCode = addrs[i].MailingPostalCode__c;
+            accsAddrs.accs[i].BillingCountry = addrs[i].MailingCountry__c;
+        }        
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update accsAddrs.accs;
+        Test.stopTest();
+        
+        // verify that we don't have new Addresses
+        list<Address__c> listAddr = [select Id from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.